docs/features/pixi_global_manifest.md

```diff
@@ -0,0 +1,222 @@
+# Pixi Global Manifest
+
+## Motivation
+
+`pixi global` is currently limited to imperatively managing CLI packages.
+The next iteration of this feature should fulfill the following needs:
+
+- Shareable global environments.
+- Managing complex environments with multiple packages as dependencies
+- Flexible exposure of binaries
+
+## Design Considerations
+
+There are a few things we wanted to keep in mind in the design:
+
+1. **User-friendliness**: Pixi is a user focused tool that goes beyond developers. The feature should have good error reporting and helpful documentation from the start.
+2. **Keep it simple**: The CLI should be all you strictly need to interact with global environments.
+3. **Unsurprising**: Simple commands should behave similar to traditional package managers.
+4. **Human Readable**: Any file created by this feature should be human-readable and modifiable.
+
+## Manifest
+
+```toml title="pixi-global.toml"
+# The name of the environment is `python`
+# It will expose python, python3 and python3.11, but not pip
+[envs.python.dependencies]
+python = { spec = "3.11.*", expose_binaries = "auto" }
+pip = "*"
+
+# The name of the environment is `python_3_10`
+# It will expose python3.10
+[envs.python_3_10.dependencies]
+python = { spec = "3.10.*", expose_binaries = { "python3.10"="python" } }
+
+```
+
+## CLI
+
+Install one or more packages `PACKAGE` into their own environments and expose their binaries.
+If no environment named `PACKAGE` exists, it will be created.
+The syntax for `MAPPING` is `exposed_name=binary_name`, so for example `python3.10=python`.
+
+```
+pixi global install [--expose MAPPING] <PACKAGE>...
+```
+
+Remove environments `ENV`.
+```
+pixi global uninstall [ENV]...
+```
+
+Upgrade all packages in environments `ENV`
+```
+pixi global upgrade [ENV]...
+```
+
+Inject package `PACKAGE` into an existing environment `ENV`.
+If environment `ENV` does not exist, it will return with an error.
+```
+pixi global inject --environment ENV PACKAGE
+```
+
+Remove package `PACKAGE` from environment `ENV`.
+If that was the last package remove the whole environment and print that information in the console.
+```
+pixi global remove --environment ENV PACKAGE
+```
+
+Update the version of one package
+```
+pixi global update --environment ENV PACKAGE
+```
+
+Set for a specific package `PACKAGE` in environment `ENV` under which `MAPPING` binaries are exposed
+The syntax for `MAPPING` is `exposed_name=binary_name`, so for example `python3.10=python`.
+```
+pixi expose --environment ENV --package PACKAGE [MAPPING]...
```

Comment:
Isnt the `--environment` argument required? Could we remove the flag name? (Also not required in pipx).

Reply:
It is required. Personally, I think required `flags` are fine.

In that specific case, making `environment` positional would be double annoying, since there can be more than one `MAPPING`. The result would be really hard to read.

Reply:
I actually agree with Bas, but to make sure we're using standards is looked at https://clig.dev/#arguments-and-flags again and they tell you flags are better over args for future proofing.

Reply:
The link didn't really put me to the right place, so here again what they say:

> Prefer flags to args. It’s a bit more typing, but it makes it much clearer what is going on. It also makes it easier to make changes to how you accept input in the future. Sometimes when using args, it’s impossible to add new input without breaking existing behavior or creating ambiguity.

Reply:
Mm I dont think I agree in this specific case but happy to follow guidelines.